audtorch/transforms/functional.py

```diff
@@ -285,7 +288,11 @@ def standardize(signal, axis=None):
     else:
         mean = np.mean(signal)
         std = np.std(signal)
-    return (signal - mean) / np.maximum(std, 1e-7)
+    if mean:
+        signal = signal - mean
+    if std:
+        signal = signal / np.maximum(std, 1e-7)
```

Comment:
I think we should only do the computation if we really have to. Fortunatly, numpy `mean` and `std` seem to allow both for `axis=None` as well. So you could change it to:
```python
if mean:
    mean_signal = np.expand_dims(np.mean(signal, axis=axis), axis=axis)
    signal = signal - mean_signal
if std:
    std_signal = np.expand_dims(np.std(signal, axis=axis), axis=axis)
    signal = signal / np.maximum(std_signal, 1e-7)
```

Note: your current implementation has a bug, as you store your calculations in the variables named `mean` and `std` which are identical with the new argument names.

Reply:
OK, I overlooked that you renamed the intermediate variables in a later commit. So there shouldn't be an issue.
But it should be possible to not check for the `axis` and only execute the calculation if really needed.

Reply:
I saw your new update and realized that you cannot implement as I proposed as `np.expand_dims` will not work with `axis=None`. So I think your current approach is the best one.